server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -491,6 +504,11 @@ public ContextIndexSearcher searcher() {
         return this.searcher;
     }
 
+    @Override
+    public ContextIndexSearcher parallelSearcher() {
+        return this.parallelSearcher;
```

Comment:
Given that out of a single search context instance, only either the parallel or the sequential search is used, it seems like a waste to create both all the time? 

I see a lot of different SearchService#createContext methods, maybe that is where we enable concurrency phase by phase?

Reply:
I was under the impression we share the search context through the phases and want several phases to allow / disallow parallel execution individually. Also, afaik creating the searcher and the resources it consumes should be relatively light-weight. Please correct me if I'm wrong on that.
Will look into the different SearchService#createContext methods as options.

Reply:
> I was under the impression we share the search context through the phases and want several phases to allow / disallow parallel execution individually.

Double check that, but we create an index searcher per shard level search at the lucene level. The index searcher is quite short-lived. In case we perform multiple queries or multiple knn searches, we do reuse the same instance, but that's the only situation. That is pretty much the same for the search context, I believe. 

> Also, afaik creating the searcher and the resources it consumes should be relatively light-weight. 

Sure, but it looks conceptually wrong to create an instance that we will never use.


Additional thought: we will have a need to tie the decision to a specific request (possibly based on info that are stored within ShardSearchRequest / features used in the SearchSourceBuilder?).